esmvaltool/diag_scripts/seaborn_diag.py

```diff
@@ -443,6 +448,12 @@ def _set_legend_title(plot_obj, legend_title: str) -> None:
         legend = plot_obj.get_legend()
     elif hasattr(plot_obj, 'legend'):  # FacetGrid, PairGrid
         legend = plot_obj.legend
+    elif isinstance(plot_obj, sns.axisgrid.JointGrid):  # JointGrid
+        # Manually create a legend if needed in JointGrid
+        handles, labels = plot_obj.ax_joint.get_legend_handles_labels()
+        if handles and labels:
+            legend = plot_obj.ax_joint.legend(handles=handles, labels=labels,
+                                              title=legend_title)
```

Comment:
It's probably good to keep this (with a small change to make sure that `legend` is always defined).

```suggestion
    elif isinstance(plot_obj, sns.axisgrid.JointGrid):  # JointGrid
        # Manually create a legend if needed in JointGrid
        handles, labels = plot_obj.ax_joint.get_legend_handles_labels()
        if handles and labels:
            legend = plot_obj.ax_joint.legend(handles=handles, labels=labels,
                                              title=legend_title)
        else:
            legend = None
```